tests/e2e/edge-middleware.test.ts

```diff
@@ -52,3 +52,20 @@ test('it should render OpenGraph image meta tag correctly', async ({ page, middl
   const size = await getImageSize(Buffer.from(imageBuffer), 'png')
   expect([size.width, size.height]).toEqual([1200, 630])
 })
+
+test('json data rewrite works', async ({ middlewarePages }) => {
+  const response = await fetch(`${middlewarePages.url}/_next/data/build-id/sha.json`, {
+    headers: {
+      'x-nextjs-data': '1',
+    },
+  })
+
+  expect(response.ok).toBe(true)
+  const body = await response.text()
+
+  expect(body).toMatch(/^{"pageProps":/)
+
+  const data = JSON.parse(body)
+
+  expect(data.pageProps.message).toBeDefined()
+})
```

Comment:
Using already existing setup in one of our fixtures https://github.com/opennextjs/opennextjs-netlify/blob/c3e328c67aee611f83567ebf95c86abcfdd52d93/tests/fixtures/middleware-pages/middleware.js#L37-L40 for middleware rewrite

and https://github.com/opennextjs/opennextjs-netlify/blob/c3e328c67aee611f83567ebf95c86abcfdd52d93/tests/fixtures/middleware-pages/pages/shallow.js#L37-L43 for asserting body